### PR TITLE
[Feature] New ACM data and a notebook for data prepare

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR include two changes:
1. Modify the ACM generation python code to include more real-like setting;
2. Add a new Notebook to introduce how to prepare the ACM data for other notebooks.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/acm_data.py [hunk 1]
```diff
@@ -88,12 +92,19 @@ def create_acm_raw_data(graph,
...
+        node_prefix_dict[ntype] = ntype[0]
+
     for ntype in graph.ntypes:
         node_dict = {}
         # generate the id column
         node_ids = graph.nodes(ntype)
-        # convert tensor to list of arrays for saving in parquet format
-        node_dict['node_id'] = convert_tensor_to_list_arrays(node_ids)
+        # pad a prefix before each node id
+        str_node_ids = np.array([f'{node_prefix_dict[ntype]}{i}' for i in node_ids.numpy()])
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply examples/acm_data.py:105]: Use the first letter is only for this specific dataset. If someone wants to change it to use other letters, they only need to change the value in the line #98, not all the lines below.

File: examples/acm_data.py [hunk 2]
```diff
@@ -124,9 +135,11 @@ def create_acm_raw_data(graph,
         edge_dict = {}
         # generate the ids columns for both source nodes and destination nodes
         src_ids, dst_ids = graph.edges(etype=(src_ntype, etype, dst_ntype))
-       # convert tensor to list of arrays for saving in parquet format
-        edge_dict['source_id'] = convert_tensor_to_list_arrays(src_ids)
-        edge_dict['dest_id'] = convert_tensor_to_list_arrays(dst_ids)
+        # pad a prefix before each node id
+        str_src_ids = np.array([f'{node_prefix_dict[src_ntype]}{i}' for i in src_ids.numpy()])
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply examples/acm_data.py:139]: ditto

File: docs/source/notebooks/Notebook_0_Data_Prepare.ipynb [hunk 3]
```diff
@@ -0,0 +1,1399 @@
...
+    "!wget -O ./acm_data.py https://github.com/awslabs/graphstorm/raw/main/examples/acm_data.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a78c88ef-91f0-4737-9506-985dc7d03c7f",
+   "metadata": {},
+   "source": [
+    "## Generate ACM Raw Table Data\n",
+    "Then we can use the command below to build the raw table data, which is the stand input data for GraphStorm's gconstruct module."
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/notebooks/Notebook_0_Data_Prepare.ipynb:92]: changed in local branch.

File: docs/source/notebooks/Notebook_0_Data_Prepare.ipynb [hunk 4]
```diff
@@ -0,0 +1,1399 @@
...
+    "!cat ./acm_raw/config.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6babcfd3-b570-4045-9cb6-541bf7ceaf4d",
+   "metadata": {},
+   "source": [
+    "#### Raw ACM Tables in the `nodes/` and `edges/` folder.\n",
+    "As defined in the `./acm_raw/config.json` file, the node data files are stored at the `./acm_raw/nodes/` folder, and edge data files are stored at the `./acm_raw/edges/` folder. General description of these files can be found at the [Input raw node/edge data files](https://graphstorm.readthedocs.io/en/latest/tutorials/own-data.html#input-raw-node-edge-data-files). Here, we can read some nodes(\"paper\") and edges(\\[\"paper\", \"citing\", \"paper\"\\]) tables to learn more about them."
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/notebooks/Notebook_0_Data_Prepare.ipynb:616]: changed in local branch and commited.

File: docs/source/notebooks/Notebook_0_Data_Prepare.ipynb [hunk 5]
```diff
@@ -0,0 +1,1399 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7237ea83-fc92-4a36-87b8-adecae9bdf99",
+   "metadata": {},
+   "source": [
+    "**The \"paper\" node table**\n",
+    "\n",
+    "The paper type node table could be read in as a Pandas DataFrame. The table has a few columns, whose name are used in the `config.json`. For the \"paper\" nodes, there is a `node_id` column, including a unique identifier for each node, a `feat` column, including a 256D numerical tensor for each node, a `text` column, including free text feature for each node, and a `label` column, including an integer to indicate the class that each node is assigned.\n",
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/notebooks/Notebook_0_Data_Prepare.ipynb:639]: changed in my local branch and commit.

File: docs/source/notebooks/Notebook_0_Data_Prepare.ipynb [hunk 6]
```diff
@@ -0,0 +1,1399 @@
...
+   "id": "8cd709f6-9089-4084-b218-71d8cf0a8924",
+   "metadata": {},
+   "source": [
+    "### GraphStorm Input Graph Data in the `./acm_gs_*p/` Folder\n",
+    "\n",
+    "In the above cells, we created a 1-partition graph in the `acm_gs_1p` folder and a 3-partition graph in the `acm_gs_3p` folder. The contents of the two folders are nearly the same, including \n",
+    "\n",
+    "1. a GraphStorm partitioned configuration JSON file;\n",
+    "2. original node id space to GraphStorm node id space mapping files, created during graph processing;\n",
+    "3. GraphStorm node id space to shuffled node id space mapping, created during graph patitioning;\n",
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/notebooks/Notebook_0_Data_Prepare.ipynb:852]: Changed in my local branch and commit.

File: docs/source/notebooks/Notebook_0_Data_Prepare.ipynb [hunk 7]
```diff
@@ -0,0 +1,1399 @@
...
+   "source": [
+    "!ls -al ./acm_gs_3p"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7cbd682e-36e8-4410-ae77-b0d6931f3052",
+   "metadata": {},
+   "source": [
+    "Because the choice of the different number of partitions, the two folders have different partition data sub-folders, named after \"part0\" to \"part*N*\", where *N* is the number of partitions specified with the `--num-parts` argument.\n",
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/notebooks/Notebook_0_Data_Prepare.ipynb:921]: Changed in my local branch and commit.

**Review by classicsong — APPROVED** [review]
LGTM

**Pull request merged**